android/src/main/java/expo/modules/xmtpreactnativesdk/XMTPModule.kt

```diff
@@ -76,16 +79,25 @@ class ReactNativeSigner(var module: XMTPModule, override var address: String) :
         continuations.remove(id)
     }
 
-    override suspend fun sign(data: ByteArray): Signature {
-        val request = SignatureRequest(message = String(data, Charsets.UTF_8))
+    override suspend fun sign(data: ByteArray): Signature? {
+        val message = String(data, Charsets.UTF_8)
+        return signLegacy(message)
+    }
+
+    override suspend fun signLegacy(message: String): Signature {
+        val request = SignatureRequest(message = message)
         module.sendEvent("sign", mapOf("id" to request.id, "message" to request.message))
         return suspendCancellableCoroutine { continuation ->
             continuations[request.id] = continuation
         }
     }
 
-    override suspend fun sign(message: String): Signature =
-        sign(message.toByteArray())
+    override fun sign(message: String): ByteArray {
+        return runBlocking {
+            signLegacy(message).toByteArray()
+        }
```

Comment:
🤔 I wonder if this will work for sure worth trying but I almost think that would just give me the bytearray of the signature object

When I think we want something more like this 
https://github.com/xmtp/xmtp-android/pull/156/files#diff-2be5e4bd7739dcb1debb7023943ee5544ebef6355405115cf89aa0e0a3ee09edR97-R102

Maybe this would do to get the bytes out of the signature 🤔 

```suggestion
        return runBlocking {
            signLegacy(message).ecdsaCompact.bytes.toByteArray()
        }
```